kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/sink/commit/Gen2Committer.scala

```diff
@@ -0,0 +1,133 @@
+package io.lenses.streamreactor.connect.aws.s3.sink.commit
+
+import io.lenses.streamreactor.connect.aws.s3.model._
+import io.lenses.streamreactor.connect.aws.s3.sink.config.S3SinkConfig
+import io.lenses.streamreactor.connect.aws.s3.sink.{IndexFileName, S3FileNamingStrategy, S3IndexNamingStrategy}
+import io.lenses.streamreactor.connect.aws.s3.storage.Storage
+import org.apache.kafka.connect.errors.ConnectException
+
+import scala.io.Source
+import scala.util.control.NonFatal
+
+class Gen2Committer(storage: Storage,
+                    fileNamingStrategyFn: Topic => S3FileNamingStrategy,
+                    bucketAndPrefixFn: Topic => BucketAndPrefix) extends Committer {
+
+  private val logger = org.slf4j.LoggerFactory.getLogger(getClass.getName)
+
+  private val indexNamingStrategy = new S3IndexNamingStrategy
```

Comment:
Style comment only:
You can define S3IndexNamingStrategy as trait and mix with the class. This way you attach the behaviour (of course making the func. protected)